### PR TITLE
cyclic: create mark tables if cyclic is enabled

### PR DESCRIPTION
**Pull request by overvenus**
### What problem does this PR solve? 

For now, we have to create mark tables manually in order to use cyclic replication.

### What is changed and how it works?

To avoid this inconvenience, `cdc cli` should create mark tables automatically when we set up a changefeed.

### Check List 

Tests 

 - Unit test
 - Integration test

Related changes

 - Need to cherry-pick to the release branch
 - Need to update the documentation

### Release note

- Improve cyclic replication set up.

### CODE REVIEW

**Comment by overvenus** [comment]
/run-integration-tests

PTAL @amyangfei @leoppro

File: cmd/client_changefeed.go [hunk 1]
```diff
@@ -181,7 +182,12 @@ func newCreateChangefeedCommand() *cobra.Command {
 					return err
 				}
 			}
-			if cyclicReplicaID != 0 && len(cyclicFilterReplicaIDs) != 0 {
+			if cyclicReplicaID != 0 || len(cyclicFilterReplicaIDs) != 0 {
+				if !(cyclicReplicaID != 0 && len(cyclicFilterReplicaIDs) != 0) {
+					return errors.Errorf(
```
Comment (amyangfei) [inline cmd/client_changefeed.go:187]:
```suggestion
					return errors.New(
```

File: cmd/client_changefeed.go [hunk 2]
```diff
@@ -224,6 +230,10 @@ func newCreateChangefeedCommand() *cobra.Command {
 					}
 				}
 			}
+			if cfg.Cyclic.IsEnabled() && !cyclic.IsTablesPaired(allTables) {
+				return errors.Errorf("normal tables and mark tables are not paired, " +
```
Comment (amyangfei) [inline cmd/client_changefeed.go:234]:
```suggestion
				return errors.New("normal tables and mark tables are not paired, " +
```

File: pkg/cyclic/mark.go [hunk 3]
```diff
@@ -14,44 +14,51 @@
...
-		}}
-	return events
+func CreateMarkTables(ctx context.Context, tables []model.TableName, upstreamDSN string) error {
+	db, err := sql.Open("mysql", upstreamDSN)
+	if err != nil {
+		return errors.Annotate(err, "Open upsteam database connection failed")
+	}
+	err = db.PingContext(ctx)
+	if err != nil {
+		return errors.Annotatef(err, "fail to open upstream TiDB connection")
```
Comment (amyangfei) [inline pkg/cyclic/mark.go:39]:
```suggestion
		return errors.Annotate(err, "fail to open upstream TiDB connection")
```

File: pkg/cyclic/mark.go [hunk 4]
```diff
@@ -14,44 +14,51 @@
...
+	}
+
+	for _, name := range tables {
+		schema, table := MarkTableName(name.Schema, name.Table)
+		_, err = db.ExecContext(ctx, fmt.Sprintf("CREATE DATABASE IF NOT EXISTS %s;", schema))
+		if err != nil {
+			return errors.Annotatef(err, "fail to create mark database")
```
Comment (amyangfei) [inline pkg/cyclic/mark.go:46]:
```suggestion
			return errors.Annotate(err, "fail to create mark database")
```

**Pull request merged**